edb/server/compiler/dbstate.py

```diff
@@ -298,6 +292,16 @@ class QueryUnit:
     def has_ddl(self) -> bool:
         return bool(self.capabilities & enums.Capability.DDL)
 
+    @property
+    def tx_control(self) -> bool:
+        return (
+            bool(self.tx_id)
+            or self.tx_rollback
+            or self.tx_commit
+            or self.tx_savepoint_declare
+            or self.tx_savepoint_rollback
+        )
```

Comment:
I'm confused... The previous commit killed it... Still, I think that refactoring this to use a simple enum is better than having like 5 or 6 separate properties.

Reply:
Sounds reasonable! I'll start a separate PR for this.